lib/fsevents-handler.js

```diff
@@ -89,6 +102,30 @@ function setFSEventsListener(path, realPath, listener, rawEmitter) {
   };
 }
 
+// Decide whether or not we should start a new higher-level
+// parent watcher
+function couldConsolidate(path) {
+  var keys = Object.keys(FSEventsWatchers);
+  var count = 0;
+
+  for (var i = 0, len = keys.length; i < len; ++i) {
+    var watchPath = keys[i];
+    if (watchPath.indexOf(path) === 0) {
+      count++;
+      if (count >= consolidateThreshhold) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+// Find the parent of a given path
+function parentOf (path) {
```

Comment:
Fixed. Sorry, didn't realize that `dirname` worked for directories parents as well.